plutus-ledger-api/test-plugin/Spec/Value/TokenName.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE BlockArguments    #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeApplications  #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+
+module Spec.Value.TokenName where
+
+import Data.ByteString (ByteString)
+import Data.Text.Encoding as TE
+import PlutusCore (DefaultFun, DefaultUni, NamedDeBruijn, someValue)
+import PlutusLedgerApi.V1.Value (TokenName (..))
+import PlutusTx (CompiledCode, getPlcNoAnn)
+import PlutusTx.Plugin.IsString (RawBytes (..), Utf8Encoded (..))
+import PlutusTx.TH (compile)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (assertBool, testCase, (@?=))
+import UntypedPlutusCore (Program (_progTerm), Term (Constant))
+
+tests :: TestTree
+tests =
+  testGroup
+    "IsString TokenName"
+    [ test_TokenName_IsString_Utf8
+    , test_TokenName_IsString_Raw
+    ]
+
+test_TokenName_IsString_Utf8 :: TestTree
+test_TokenName_IsString_Utf8 = testCase "Utf8Encoded" do
+  term $$(compile [||utf8Encoded "Имя Токена" :: TokenName||])
```

Comment:
GHC Core is:
```haskell
fromString
  @(Utf8Encoded TokenName)
  ($fIsStringUtf8Encoded
     @TokenName
     (MkCoercible
        @Type
        @TokenName
        @BuiltinByteString
        @~(N:TokenName[0] :: TokenName ~R# BuiltinByteString)))
  (unpackCStringUtf8#
     "\\208\\152\\208\\188\\209\\143 \\208\\162\\208\\190\\208\\186\\208\\181\\208\\189\\208\\176"#)
```

Reply:
So essentially what the plugin now does is it looks for the `Utf8Encoded` constructor and transforms this piece of code to basically the following:
```
fromString
  @(Utf8Encoded TokenName)
  (unpackCStringUtf8# "\\208\\152\\208\\188\\209\\143 \\208\\162\\208\\190\\208\\186\\208\\181\\208\\189\\208\\176")
```

Is that correct?